src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -712,10 +715,15 @@ public static long onSleepFinished(ServerLevel level, long newTime, long minTime
         return fire(new SleepFinishedTimeEvent(level, newTime, minTime)).getNewTime();
     }
 
+    @Deprecated(forRemoval = true, since = "1.21.4")
     public static List<PreparableReloadListener> onResourceReload(ReloadableServerResources serverResources, RegistryAccess registryAccess) {
         return fire(new AddReloadListenerEvent(serverResources, registryAccess)).getListeners();
     }
 
+    public static List<PreparableReloadListener> onResourceReload(ReloadableServerResources serverResources, HolderLookup.Provider lookupProvider, @Deprecated(forRemoval = true, since = "1.21.4") RegistryAccess registryAccess) {
+        return fire(new AddReloadListenerEvent(serverResources, lookupProvider, registryAccess)).getListeners();
```

Comment:
Looks fine, only comment is that we typically add new parameters to the end of a function not the middle.
But this will work.

Reply:
Noted for next time.